src/main/resources/com/google/api/codegen/csharp/gapic_resourcenames.snip

```diff
@@ -292,13 +292,13 @@
         public override string ToString() => Name.ToString();
 
         /// <inheritdoc />
-        public override int GetHashCode() => ToString().GetHashCode();
+        public override int GetHashCode() => Name.GetHashCode();
 
         /// <inheritdoc />
         public override bool Equals(object obj) => Equals(obj as {@resource.typeName});
 
         /// <inheritdoc />
-        public bool Equals({@resource.typeName} other) => ToString() == other?.ToString();
+        public bool Equals({@resource.typeName} other) => Name.Equals(other?.Name);
```

Comment:
I don't think we want to do this part yet - because we still want an "unknown" which is actually a ProjectName to match a *real* ProjectName, if they have the same string representation. This will take a bit more work.